paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsActivity.kt

```diff
@@ -76,20 +78,26 @@ internal class PaymentOptionsActivity : BaseSheetActivity<PaymentOptionResult>()
             closeSheet(it)
         }
 
-        viewModel.error.launchAndCollectIn(this) { error ->
-            updateErrorMessage(
-                messageView,
-                error?.let { BaseSheetViewModel.UserErrorMessage(it) }
-            )
-        }
-
         viewBinding.contentContainer.setContent {
             StripeTheme {
                 val currentScreen by viewModel.currentScreen.collectAsState()
                 currentScreen.PaymentOptionsContent(starterArgs)
             }
         }
 
+        viewBinding.message.setContent {
+            StripeTheme {
+                val errorMessage by viewModel.error.collectAsState(initial = null)
+
+                errorMessage?.let { error ->
+                    ErrorMessage(
+                        error = error,
+                        modifier = Modifier.padding(vertical = 2.dp, horizontal = 20.dp),
```

Comment:
This is to keep things as-is from a UI perspective. We’ll make this nicer in a follow-up.